hooks/python-hooks/apigw-enforce-authorizer/src/awssamples_apigwenforceauthorizer_hook/handlers.py

```diff
@@ -0,0 +1,112 @@
+import logging
+import resource
+from typing import Any, Dict, MutableMapping, Optional
+
+from cloudformation_cli_python_lib import (
+    BaseHookHandlerRequest,
+    HandlerErrorCode,
+    Hook,
+    HookInvocationPoint,
+    OperationStatus,
+    ProgressEvent,
+    SessionProxy,
+    exceptions,
+)
+
+from .models import HookHandlerRequest, TypeConfigurationModel
+
+# Use this logger to forward log messages to CloudWatch Logs.
+LOG = logging.getLogger(__name__)
+TYPE_NAME = "AWSSamples::APIGWEnforceAuthorizer::Hook"
+
+hook = Hook(TYPE_NAME, TypeConfigurationModel)
+test_entrypoint = hook.test_entrypoint
+
+PUBLIC_METHODS = ["options"]
+
+@hook.handler(HookInvocationPoint.CREATE_PRE_PROVISION)
+def pre_create_handler(
+        session: Optional[SessionProxy],
+        request: HookHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+        type_configuration: TypeConfigurationModel
+) -> ProgressEvent:
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.FAILED
+    )
+    target_name = request.hookContext.targetName
+    target_model = request.hookContext.targetModel
+    resource_properties = target_model.get("resourceProperties") if target_model is not None else None
+
+    if validate_auth(target_name, resource_properties):
+        progress.status = OperationStatus.SUCCESS
+    else:
+        progress.status = OperationStatus.FAILED
+        progress.message = "Not all paths and methods have authorizers specified."
+    return progress
+
+
+@hook.handler(HookInvocationPoint.UPDATE_PRE_PROVISION)
```

Comment:
It seems that `pre_create_handler()` and `pre_update_handler()`, at this time, have the same code and behavior.  Can you please consider the following:
- move the decorator: `@hook.handler(HookInvocationPoint.UPDATE_PRE_PROVISION)` above, before the `@hook.handler(HookInvocationPoint.CREATE_PRE_PROVISION)` decorator in `pre_create_handler()`, and
- remove `pre_update_handler()` below.
Thank you!

Reply:
Moved both decorators to the same handler and removed the duplicated one.